copier/vcs.py

```diff
@@ -68,23 +68,30 @@ def get_repo(url: str) -> OptStr:
             - git+https://mywebsiteisagitrepo.example.com/
             - /local/path/to/git/repo
             - /local/path/to/git/bundle/file.bundle
+            - ~/path/to/git/repo
+            - ~/path/to/git/repo.bundle
     """
     for pattern, replacement in REPLACEMENTS:
         url = re.sub(pattern, replacement, url)
+
     url_path = Path(url)
-    if not (
-        url.endswith(GIT_POSTFIX)
-        or url.startswith(GIT_PREFIX)
-        or is_git_repo_root(url_path)
-        or is_git_bundle(url_path)
-    ):
+    if url.startswith("~"):
+        url_path = url_path.expanduser()
+
+    is_git_url = url.endswith(GIT_POSTFIX) or url.startswith(GIT_PREFIX)
+    is_local_path = is_git_repo_root(url_path) or is_git_bundle(url_path)
+
+    if not is_git_url and not is_local_path:
         return None
 
-    if url.startswith("git+"):
-        url = url[4:]
-    elif url.startswith("https://") and not url.endswith(GIT_POSTFIX):
-        url = "".join((url, GIT_POSTFIX))
-    return url
+    if is_git_url:
+        if url.startswith("git+"):
+            url = url[4:]
+        elif url.startswith("https://") and not url.endswith(GIT_POSTFIX):
+            url = "".join((url, GIT_POSTFIX))
+        return url
+
+    return url_path.as_posix()
```

Comment:
Previously, `url_path` was never returned, and we needed to return it to make the rest of the flow work.
I've refactored a bit to make things more readable, and to return either `url` or `url_path` depending on whether the URL is a Git URL or a local path.

Reply:
Refactored again to avoid computing unnecessary things.